reinforcement_learning/rlclientlib/logger/event_batcher.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include "event_queue.h"
+#include "ranking_event.h"
+#include "utility/data_buffer.h"
+
+namespace reinforcement_learning {
+  namespace utility {
+    class event_batcher {
+    public:
+      event_batcher();
+      void batch_serialize(data_buffer& oss, size_t& remaining, event_queue<ranking_event>& queue, size_t _send_high_water_mark);
+      void batch_serialize(data_buffer& oss, size_t& remaining, event_queue<outcome_event>& queue, size_t _send_high_water_mark);
+
+      template<typename TEvent>
+      void batch_serialize(data_buffer& oss, size_t& remaining, event_queue<TEvent>& queue, size_t _send_high_water_mark);
+
+    private:
+      size_t _batch_size = 400;
+      template<typename TEvent, typename TResultObject, typename TResultObjectBuilder>
+      void batch_serialize_internal(data_buffer& oss, size_t& remaining, event_queue<TEvent>& queue, size_t _send_high_water_mark, size_t batch_size);
+    };
+
+    template<typename TEvent>
+    void event_batcher::batch_serialize(data_buffer& oss, size_t& remaining, event_queue<TEvent>& queue, size_t _send_high_water_mark)
+    {
+      // This is used by unit_test
```

Comment:
Why do we need this method?

Reply:
The problem is async_batcher actually is designed to support any events. Unit tests are defining different events contains of string data, which does not have a flatbuffer schema.

---
In reply to: [228610098](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1653#discussion_r228610098) [](ancestors = 228610098)

Reply:
1) I think it is. async_batcher only providers numbers of queue to event_batcher. EventBatcher is the flatbuffer serialization.
2) It might not be needed now. Rajan will have flatbuffer preamble header soon. I don't see a point to have flatbuffer to replace all unit tests now.

---
In reply to: [228641030](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1653#discussion_r228641030) [](ancestors = 228641030)